### ISSUE TITLE
Rethink API

### ISSUE THREAD

**Issue by 3lvis** [hyperoslo/Sync#37]
I've been wondering what if we changed the API to something more simple.

# New
```objc
// [Sync changes:notes entityName:@"Notes" dataStack:self.dataStack completion:nil];

+ (void)changes:(NSArray *)changes
     entityName:(NSString *)entityName
      dataStack:(DATAStack *)dataStack
     completion:(void (^)(NSError *error))completion;
```

# Old
```objc
// [Sync processChanges:notes usingEntityName:@"Notes" dataStack:self.dataStack completion:nil];

+ (void)processChanges:(NSArray *)changes
       usingEntityName:(NSString *)entityName
             dataStack:(DATAStack *)dataStack
            completion:(void (^)(NSError *error))completion;
```

**Comment by jgorset** [comment]
Coming from Ruby, I like how the old one reads like a sentence. Maybe you could still do that with less verbosity, though.

**Comment by jgorset** [comment]
Actually the first argument of the proposed syntax reads better (`Sync changes [...]`), but the second worse (`[...] entityName`).

**Comment by jgorset** [comment]
So I'm totally bad at this, but what about…

```objective-c
[Sync changes:notes as:@"Notes" [...]
```

**Comment by jgorset** [comment]
Hell, if DATAStack does what I think it does, what about…

```objective-c
[Sync changes:notes as:@"Notes" on:self.dataStack ...]
```

…? Too crazy?

**Comment by 3lvis** [comment]
>Hell, if DATAStack does what I think it does, what about…
>
>```objective-c
>[Sync changes:notes as:@"Notes" on:self.dataStack ...]
>```
>
>…? Too crazy!?

That's very Swift of you, we should totally change the API once we make a Swift bridge.

**Comment by jgorset** [comment]
So ahead of my time, you know. ;-)

**Comment by RamonGilabert** [comment]
What about

```objc
+ (void)changes:(NSArray *)changes
    inEntityName:(NSString *)entityName
         dataStack:(DATAStack *)dataStack
        completion:(void (^)(NSError *error))completion;
```

**Comment by 3lvis** [comment]
@RamonGilabert I like that!

**Comment by RamonGilabert** [comment]
@3lvis I was thinking in adding just `changes inEntity dataStack` (or just stack) the user will see the placeholder saying, `entityName` so I think they'll know what to put.

**Comment by 3lvis** [comment]
@RamonGilabert I haven't choose that direction because of NSEntityDescription, in a Core Data context when you say entity most of the time you are referring to a `NSEntityDescription` instance.

We could go for 

```objc
+ (void)changes:(NSArray *)changes
   inEntityName:(NSString *)entityName
      dataStack:(DATAStack *)dataStack
     completion:(void (^)(NSError *error))completion
```

**Comment by RamonGilabert** [comment]
All right, let me finish a small surprise and I'll go for it! :dancers:

**Comment by 3lvis** [comment]
Fixed on #40!

Thanks @RamonGilabert :star2:  :bouquet: :star2:
